data/clean.py

```diff
@@ -0,0 +1,109 @@
+# import csv
+import pandas
+import re
+from pandas import DataFrame, isna
+from dataclasses import dataclass, field
+from collections import defaultdict
+from pprint import pprint
+import numpy as np
+from typing import Dict, List
+
+number_regex = r"-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?"
+range_regex = f"{number_regex}(?:-{number_regex})?"
+unit_regex = r"[^\s]*"
+
+value_regex = re.compile(f"(?:(.*):)?\s*({range_regex})\s*({unit_regex})")
```

Comment:
I'm think (and I could be wrong!) that value_regex won't work on cells where we have multiple variants, i.e. "Avg: 1.5e3m, Max: 5.1e3m". So we could either (significantly?) complicate the regex, or separate this into two different columns for average depth and max depth so no cell has a string descriptor followed by a colon.

Reply:
That splitting is done elsewhere, see line 69 (nice)

Reply:
Also I'm outputting the data as a tab-separated value rather than a comma-separated one so it's easier to do comma separation within the spreadsheet itself